firefox-ios/Client/Frontend/Autofill/AutofillAccessoryView.swift

```diff
@@ -0,0 +1,112 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+import Common
+import Shared
+
+/// Custom UIBarButtonItem with an autofill accessory view.
+///
+/// This class provides a reusable and configurable autofill accessory view for use in navigation bars.
+///
+/// # Attributes
+/// - `accessoryImageViewSize`: Size of the accessory image view.
+/// - `accessoryButtonStackViewSpacing`: Spacing between the accessory image view and text label in the stack view.
+/// - `cornerRadius`: Corner radius for the accessory view.
+///
+/// # Properties
+/// - `accessoryImageViewTintColor`: Tint color for the accessory image view.
+/// - `backgroundColor`: Background color for the accessory view.
+///
+/// # Methods
+/// - `init(image:labelText:tappedAction:)`: Initializes the accessory view with an image, label, and optional tap action.
+/// - `tappedAccessoryButton()`: Handles the tap action on the accessory view.
+class AutofillAccessoryView: UIBarButtonItem {
+    // MARK: - Constants
+    private struct UX {
+        static let accessoryImageViewSize: CGFloat = 24
+        static let accessoryButtonStackViewSpacing: CGFloat = 2
+        static let cornerRadius: CGFloat = 4
+        static let fontSize: CGFloat = 16
+    }
+
+    // MARK: - Properties
+    private let accessoryImageView: UIImageView
+    private let useAccessoryTextLabel: UILabel
+    private let tappedAccessoryButtonAction: (() -> Void)?
+
+    /// Tint color for the accessory image view.
+    var accessoryImageViewTintColor: UIColor? {
+        get {
+            return accessoryImageView.tintColor
+        }
+        set {
+            accessoryImageView.tintColor = newValue
+        }
+    }
```

Comment:
what do you think about making this view `ThemeApplicable` and setting the `tintColor` in `applyTheme` so that we're following this consistent pattern throughout the app?

Reply:
This is not a VC ...